spec/checkMergeConflictSpec.js

```diff
@@ -0,0 +1,261 @@
+        labels: [mergeConflictLabel.name],
+      };
+      expect(github.issues.addLabels).toHaveBeenCalledWith(params);
+    });
+  });
+
+  describe('pull request merge does not result in merge conflict', () => {
+    beforeEach(async () => {
+      // Simulate a merged pull request.
+      payloadData.payload.pull_request.merged = true;
+      payloadData.payload.action = 'closed';
+
+      const pullRequest = { ...payloadData.payload.pull_request };
+      pullRequest.merged = false;
+      pullRequest.mergeable = true;
+
+      github.pulls = {
+        get: jasmine.createSpy('get').and.resolveTo({
+          data: pullRequest,
+        }),
+        list: jasmine.createSpy('list').and.resolveTo({
+          data: [pullRequest],
+        }),
+      };
+
+      await robot.receive(payloadData);
+    });
+
+    it('checks for merge conflict', () => {
+      expect(
+        checkMergeConflictModule.checkMergeConflictsInAllPullRequests
+      ).toHaveBeenCalled();
+      expect(
+        checkMergeConflictModule.checkMergeConflictsInPullRequest
+      ).toHaveBeenCalled();
+    });
+
+    it('does not ping pr author', () => {
+      expect(github.issues.createComment).not.toHaveBeenCalled();
+    });
+    it('does not add merge conflict label', () => {
+      expect(github.issues.addLabels).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('pull request already has merge conflict label', () => {
+    beforeEach(async () => {
+      // Simulate synchronize event.
+      payloadData.payload.action = 'synchronize';
+
+      const pullRequest = { ...payloadData.payload.pull_request };
+      pullRequest.merged = false;
+      pullRequest.mergeable = true;
+      pullRequest.labels.push(mergeConflictLabel);
+
+      github.pulls = {
+        get: jasmine.createSpy('get').and.resolveTo({
+          data: pullRequest,
+        }),
+        list: jasmine.createSpy('list').and.resolveTo({
+          data: [pullRequest],
+        }),
+      };
+
+      await robot.receive(payloadData);
+    });
+
+    it('should check for merge conflict', () => {
+      expect(
+        checkMergeConflictModule.checkMergeConflictsInPullRequest
+      ).toHaveBeenCalled();
+    });
+
+    it('does not ping pr author', () => {
+      expect(github.issues.createComment).not.toHaveBeenCalled();
+    });
+
+    it('does not add merge conflict label', () => {
+      expect(github.issues.addLabels).not.toHaveBeenCalled();
+    });
+
+    it('removes merge conflict label', () => {
+      expect(github.issues.removeLabel).toHaveBeenCalled();
+      const params = {
+        repo: payloadData.payload.repository.name,
+        owner: payloadData.payload.repository.owner.login,
+        number: payloadData.payload.pull_request.number,
+        name: mergeConflictLabel.name,
+      };
+      expect(github.issues.removeLabel).toHaveBeenCalledWith(params);
+    });
+  });
+
+  describe('pull request already has merge conflict label and merge conflict isnt fixed', () => {
```

Comment:
pull request already has merge conflict label and merge conflict isnt fixed -> pull request with unresolved merge conflicts has merge conflict label

Reply:
Done.